benchmarks/rewriting.py

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env python3
+"""Benchmarks for the pattern rewriter of the xDSL implementation."""
+
+from typing import cast
+
+from benchmarks.workloads import WorkloadBuilder
+from xdsl.context import Context
+from xdsl.dialects.arith import (
+    AddiOp,
+    Arith,
+    ConstantOp,
+    SignlessIntegerBinaryOperationHasCanonicalizationPatternsTrait,
+    SignlessIntegerBinaryOperationWithOverflow,
+    SubiOp,
+)
+from xdsl.dialects.builtin import Builtin, IntegerAttr, IntegerType, ModuleOp
+from xdsl.ir import Region
+from xdsl.ir.post_order import PostOrderIterator
+from xdsl.irdl import OpDef, VarIRConstruct, get_variadic_sizes, traits_def
+from xdsl.parser import Parser as XdslParser
+from xdsl.pattern_rewriter import PatternRewriter, Worklist
+from xdsl.rewriter import InsertPoint
+from xdsl.traits import (
+    Commutative,
+    HasCanonicalizationPatternsTrait,
+    MemoryEffect,
+    Pure,
+)
+from xdsl.transforms.canonicalization_patterns.utils import const_evaluate_operand
+from xdsl.transforms.canonicalize import CanonicalizePass
+from xdsl.transforms.dead_code_elimination import (
+    LiveSet,
+    is_trivially_dead,
+    result_only_effects,
+    would_be_trivially_dead,
+)
+
+CTX = Context(allow_unregistered=True)
+CTX.load_dialect(Arith)
+CTX.load_dialect(Builtin)
+
+CANONICALIZE_PASS = CanonicalizePass()
+
+
+def parse_module(context: Context, contents: str) -> ModuleOp:
+    """Parse a MLIR file as a module."""
+    parser = XdslParser(context, contents)
+    return parser.parse_module()
+
+
+class ConstantFolding:
+    """Benchmark rewriting in xDSL."""
+
+    WORKLOAD_CONSTANT_20 = parse_module(CTX, WorkloadBuilder.constant_folding(20))
+    WORKLOAD_CONSTANT_100 = parse_module(CTX, WorkloadBuilder.constant_folding(100))
+    WORKLOAD_CONSTANT_1000 = parse_module(CTX, WorkloadBuilder.constant_folding(1_000))
```

Comment:
Does this have to be hardcoded here? Can it be dynamic (i.e., configured as such down in `main`?).
I have the feeling that I've asked about this before, but I might be wrong :)